README.md

```diff
@@ -1,2 +1,129 @@
+> Using Workload Identity allows you to assign distinct, fine-grained identities and authorization for each application in your cluster.
+
+
+## Prerequisites
+1. Install [gcloud](https://cloud.google.com/sdk/docs/install)
+2. A cluster on GCP. [Creating a cluster ](https://github.com/giantswarm/capo-mc-bootstrap/)
+
+
+## Usage
+
+#### 1. Enable workload identity on your cluster
+
+Workload identity being enabled on this cluster. Note, when you enable Workload Identity on a cluster, GKE automatically creates a fixed workload identity pool for the cluster's Google Cloud project.
+A workload identity pool allows IAM to understand and trust Kubernetes service account credentials.
+The workload identity pool has the following format:
+```
+  PROJECT_ID.svc.id.goog
+```
+
+#### 2. Register your cluster
+```
+export CLUSTER_NAME="<insert-cluster-name-here>" 
+export MEMBERSHIP_NAME="$CLUSTER_NAME-workload-identity"
+export GCP_PROJECT_NAME="<insert-project-name-here>"
+export KUBECONFIG_CONTEXT="<insert-context-here>"
+export KUBECONFIG_PATH="~/.kube/config"
+```
+```
+gcloud container hub memberships register "$MEMBERSHIP_NAME" \
+    --project "$GCP_PROJECT_NAME" \
+    --context="$KUBECONFIG_CONTEXT" \
+    --kubeconfig="$KUBECONFIG_PATH" \
+    --enable-workload-identity \
+    --has-private-issuer
+```
+
+#### 3. Configuration
+
+##### 3.1 Create a kubernetes service account
+
+```
+export KUBE_SA_NAME="<inset-sa-name-here>"
+export KUBE_NAMESPACE="<insert-namespace-here>"
+
+kubectl create serviceaccount $KUBE_SA_NAME \
+    --namespace $KUBE_NAMESPACE
+```
+
+##### 3.2 [Create a GCP service account](https://cloud.google.com/iam/docs/creating-managing-service-accounts#creating) or 
+
+```
+export GOOGLE_SA_NAME="<insert-gcp-service-account-name-here>"
+gcloud iam service-accounts create "$GOOGLE_SA_NAME" --project="$GCP_PROJECT_NAME"
+```
+
+##### 3.3 Grab important information that you'll need. 
+These are the following:
+  * Your workload identity pool id
+  * Your identity provider 
+The above can be obtained from the output of the command below
+```
+gcloud container hub memberships describe $MEMBERSHIP_NAME
+```
+
+##### 3.4 Ensure that your gcp service account has the roles that you need. 
+
+```
+  export GOOGLE_SA_ID="$GOOGLE_SA_NAME@$GCP_PROJECT_NAME.iam.gserviceaccount.com"
+
+  # this policy binding associates the GCP service account with a Kubernetes service account
+  gcloud iam service-accounts add-iam-policy-binding \
+    --project "$GCP_PROJECT_NAME" \
+    "$GOOGLE_SA_ID" \
+    --role=roles/iam.workloadIdentityUser \
+    --member="serviceAccount:$WORKLOAD_ID_POOL[$KUBE_NAMESPACE/$KUBE_SA_NAME]"
+```
+
+##### 3.5 Add computer viewer permissions
+```
+  # give the service account permissions to view compute resources
+  gcloud projects add-iam-policy-binding "$GCP_PROJECT_NAME" \
+    --role=roles/compute.viewer \
+    --member="serviceAccount:$GOOGLE_SA_ID"
+```
```

Comment:
This isn't exactly **necessary**. I would rather say `Add necessary permissions to the GCP Service Account` and specify that these should be the permissions necessary for the workload to function. Maybe keep the `compute.viewer` as an example

Reply:
Good catch 👊🏾